### ISSUE TITLE
Make -no-hooks default with execute-keys and evaluate-commands

### ISSUE THREAD

**Issue by danr** [mawww/kakoune#1747]
This is a follow-up on #1692 where it was identified that hooks enabled being the default on exec and eval is a surprising default. Personally I typically add `-no-hooks` as often as I can remember to exec and eval and I can count without using any hands when I actually wanted to have them on.

This issue suggests that hooks should be off by default. They could be enabled with a `-with-hooks` flag.

**Comment by mawww** [comment]
Yeah, I think such a change would make sense, but we need to experiment with it to see what it breaks.

**Comment by lenormf** [comment]
Related #707.

**Comment by mawww** [comment]
I have been experimenting with this, and it is not as good as expected.

It seems the root of the problem is that we try to treat eval and exec the same way. But typically, we want hooks active in most `eval`, in particular any `edit` command should run with hooks, except when we are sure we do not want hooks to run. `exec` are often not expected to trigger hooks.

Also, some hooks really want to always run, such as the ones taking care of removing temporary folders when a fifo is closed. We do not want to leave temporary directories because a fifo was closed during a non-hooked eval. This is why the `hook` command gained a `-always` switch.

Finally, the initial issue would now be fixed because faces, and hence colorschemes, are now scoped, meaning those hooks would be less likely to interfere.

All that to say, I am still unsure what the best solution would be, but while disabling hooks by default felt like a nice idea, it proved to be unsufficient.
